src/main/java/bio/terra/service/resourcemanagement/ResourceService.java

```diff
@@ -236,6 +237,22 @@ public GoogleBucketResource lookupBucketMetadata(String bucketResourceId) {
     return bucketService.getBucketResourceById(UUID.fromString(bucketResourceId), false);
   }
 
+  /**
+   * Fetch an existing storage_account metadata row. Note this method does not check for the
+   * existence of the underlying cloud resource. This method is intended for places where an
+   * existence check on the associated cloud resource might be too much overhead (e.g. DRS lookups).
+   * Most storage account lookups should use the lookupStorageAccount method instead, which has
+   * additional overhead but will catch metadata corruption errors sooner.
+   *
+   * @param storageAccountId our identifier for the storage account
+   * @return a reference to the storage account as a POJO AzureStorageAccountResource
+   * @throws AzureResourceNotFoundException if the storage_account metadata row does not exist
+   */
+  public AzureStorageAccountResource lookupStorageAccountMetadata(String storageAccountId) {
```

Comment:
Is there a reason to not add `azure` and `gcs` to the method names that are specific to those platforms? From reading the method names I wouldn't assume that `lookupBucketMetadata` was gcs specific, for example.

Reply:
Yeah, I left those out in the interest of keeping the methods a little shorter since they're starting to get really long :)